### DISCUSSION TITLE
Buffer to short when adding an indices accessor for a primitive

### DISCUSSION THREAD

**Discussion by Rennzie** [donmccurdy/glTF-Transform#445] [Q&A] [answered]
I've run into an issue trying to reprocess a `LINES` primitive to use indices instead of duplicate positions. I've removed the dup verts from the positions accessor and created a set of indices. I created a new accessor which I set to the primitive. I've reused the buffer referenced by the positions primitive when creating the indices accessor. The code looks like this:

```typescript
const accessor = doc
      .createAccessor()
      .setBuffer(positions.getBuffer())
      .setType(AccessorTypes.SCALAR)
      .setArray(new Uint16Array(newIndices))
      .setName('indices');

linePrimitive.setIndices(accessor);
```

I then export the document as a `.glb` and run it through `gltfpack`. This errors out with `Error: Error loading input.glb: buffer too short`.

I've also tried adding a new buffer but when it comes to writing the `.glb` it errors with `Error: GLB must have 0–1 buffers.`.

Is there something obvious I'm missing? Or is there a way to increase the size of the underlying buffer or merge two buffers before writing?

Any help would be greatly appreciated.

**Comment by donmccurdy** [comment, marked as answer]
The library includes a `weld()` function for computing indices, although I'm not sure I've tested it with `LINES` primitives yet. Could you try this?

```js
import { weld } from '@gltf-transform/functions';

await document.transform(
  weld({tolerance: 1e-4})
);
```

If not, it might also be worth trying `io.setVertexLayout(VertexLayout.SEPARATE)` (before exporting the model) to see if that fixes whatever buffer layout issue is going on.

If none of that works, would you be able to share a sample input file?

**Reply by Rennzie** [reply]
Thanks for the quick response @donmccurdy. I'll give those a try and let you know how it goes.

**Reply by Rennzie** [reply]
Hey @donmccurdy, I tested the weld on my `input.glb`. Happy to report that it works great on `LINE` primitives. 

Turns out the issue was something different. In the process of creating indices, I was also filtering the line through a simplification process. I hadn't been handling the index cleanup properly and that was resulting in weird accessors which caused the error in `gltfpack`. 

Reading your `weld` code taught me a lot and I've managed to fix my process. Thanks for the awesome lib.
